### ISSUE TITLE
ProcProxy does not work

### ISSUE THREAD

**Issue by mitnk** [xonsh/xonsh#2179]
With following code in xonshrc:
```
def foo():
    return "Banana for scale.\n"
foo.__xonsh_threadable__ = False
aliases['ffoo'] = foo
```
Then run
```
$ ffoo
```
It will stuck.

And is there any other built-in alias using `ProcProxy`, which I think should also have this issue?

P.S. another thing I notice that I cannot name this alias to `foo`, it will be the function object, rather than a alias call.

**Comment by scopatz** [comment]
Thanks for reporting @mitnk.

> P.S. another thing I notice that I cannot name this alias to foo, it will be the function object, rather than a alias call.

This is actually by design. Xonsh has a "Python always wins" philosophy. That means that names in the Python execution context take precedence over subprocess names.

**Comment by scopatz** [comment]
Hi @mitnk, this was a tricky one but should have a fix in #2180, if you want to check it out.
